### ISSUE TITLE
Warn the user when creating a project which has access to Argo CD itself

### ISSUE THREAD

**Issue by crenshaw-dev** [argoproj/argo-cd#8982]
# Summary

If the user tries to create a project that has access to the namespace where Argo CD is installed, we should present some kind of warning.

# Motivation

Argo CD admins should be very selective about who gets access to modify the Argo CD resources. It might be too easy for an admin to click through the project creation flow without considering whether the access is too broad. Granting access to the Argo CD namespace is a special case, because it could potentially allow reconnaissance by threat actors (reading RBAC policies) or complete Argo CD hijacking (editing RBAC policies).

# Proposal

Both the UI and the CLI should show a warning when a project is created or modified indicating that the project has some kind of access to the namespace where Argo CD is installed. Ideally that warning would link to somewhere in the docs which explains why that access could be problematic.

**Comment by rishabh625** [comment]
Warning should be only in case if project creation destination is * or argocd installed namespace?

**Comment by crenshaw-dev** [comment]
@rishabh625 exactly!

**Comment by nastacio** [comment]
I agree with the warning. My biggest concern, at the Argo CD level, is that any application following the [App-Of-Apps pattern](https://argo-cd.readthedocs.io/en/stable/operator-manual/cluster-bootstrapping/#app-of-apps-pattern) needs access to the namespace of the argocd instance because [Argo CD only recognizes Applications added to that namespace](https://github.com/argoproj/argo-cd/issues/3847).

Unless I am missing something, this model seems fundamentally broken.

**Comment by crenshaw-dev** [comment]
@nastacio you're not missing anything. App of Apps and ApplicationSet should currently be thought of as fundamentally admin-level tools. If/when `project` restriction propagation is implemented, that level of caution can be relaxed.

We're [considering](https://github.com/argoproj/argo-cd/pull/9016#discussion_r844369097) project restriction propagation for ApplicationSets.

**Comment by nastacio** [comment]
Doesn't it follow that Argo admins must also own the contents of the gitrepos allowed in the AppProject, lest someone also gain access to manage project-whitelisted resources inside the namespace of the Argo CD instance?

**Comment by crenshaw-dev** [comment]
I think it does.

In other words, for any AppProject which has access to resources in the argocd namespace, only admins should be allowed to push to the repositories allowed for that AppProject.

**Comment by nastacio** [comment]
>for any AppProject which has access to resources in the argocd namespace

That would be all AppProjects containing an App-of-Apps.

Our org is looking into centrally managing an Argo control plane for various service teams owning their own Applications, so it is looking like I need to give each team an Argo CD instance "defanged" of service account roles that allows the instance to access anything at cluster level.

**Comment by nastacio** [comment]
Have we considered design alternatives that would block allow Application resources from accessing the namespace of the Argo CD instance.

For instance, things like the Argo CD instance acknowledging Application resources in different namespaces or not requiring AppProject to have the destination of the Argo CD instance (local cluster + instance namespace) in the "destinations" field?

If we can guarantee that Applications only have access to the remaining destinations in the AppProject, then there is no need to worry as much about Argo admins owning every Git repository added to the project.

**Comment by crenshaw-dev** [comment]
I'm not sure I understand your suggestion(s). Are these ideas specific to securing apps of apps, or just apps in general?

**Comment by nastacio** [comment]
The idea is for apps in general.

If Argo did not allow resources in an application to reference the namespace of the Argo CD instance, there would be no need to worry about non-admins having control of the source Git repositories.

**Comment by crenshaw-dev** [comment]
Ah yeah, agreed. For people who don't bootstrap their Argo CD install, that would be great. For others, they'd still need one project which can manage the Argo CD namespace.

**Comment by crenshaw-dev** [comment]
Related: https://github.com/argoproj/argo-cd/issues/2785
